source/routing/tests/circle_test.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright © 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE circle_test
+
+#include "type/geographical_coord.h"
+#include "routing/circle.h"
+#include <boost/test/unit_test.hpp>
+#include <iomanip>
+#include <vector>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <stdlib.h>
+#include <time.h>
+
+
+using namespace navitia::routing;
+
+BOOST_AUTO_TEST_CASE(project_in_direction_test) {
+    navitia::type::GeographicalCoord coord_Paris = navitia::type::GeographicalCoord(2.3522219000000177,48.856614);
+    navitia::type::GeographicalCoord coord_North = navitia::type::GeographicalCoord(90,0);
+    navitia::type::GeographicalCoord coord_Equator = navitia::type::GeographicalCoord(0,48.856614);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 80, 10).distance_to(coord_Paris), 10, 0.1);
```

Comment:
since the important stuff there is that when you project, the point is at the right distance, it might be easier to read with loops and a list of distances